src/Diff.re

```diff
@@ -28,25 +28,58 @@ module MakeDiff = (IO1: ImageIO.ImageIO, IO2: ImageIO.ImageIO) => {
       let row = IO1.readRow(base, y);
       let row2 = IO2.readRow(comp, y);
 
+      let getBaseColor = (x, rowToGet) => {
+        IO1.readImgColor(
+          x,
+          rowToGet == y ? row : IO1.readRow(base, rowToGet),
+          base,
+        );
+      };
+
+      let getCompColor = (x, rowToGet) => {
+        IO2.readImgColor(
+          x,
+          rowToGet == y ? row2 : IO2.readRow(comp, rowToGet),
+          comp,
+        );
+      };
+
       for (x in 0 to base.width - 1) {
         if (x >= comp.width || y >= comp.height) {
           let (_r, _g, _b, a) = IO1.readImgColor(x, row, base);
           if (a != 0) {
             countDifference(x, y);
           };
         } else {
-          let (r, g, b, a) = IO1.readImgColor(x, row, base);
-          let (r1, g1, b1, a1) = IO2.readImgColor(x, row2, comp);
+          let baseColor = IO1.readImgColor(x, row, base);
+          let compColor = IO2.readImgColor(x, row2, comp);
 
-          if (r != r1 || g != g1 || b != b1 || a != a1) {
+          if (!Helpers.isSameColor(baseColor, compColor)) {
             let delta =
-              ColorDelta.calculatePixelColorDelta(
-                (r, g, b, a),
-                (r1, g1, b1, a1),
-              );
+              ColorDelta.calculatePixelColorDelta(baseColor, compColor);
 
             if (delta > maxDelta) {
-              countDifference(x, y);
+              let isAntialiased =
```

Comment:
I feel that there should be some max value when we should check for anti-aliasing. For example, if you have 255, 255, 255, and 0,0,0 – it is definetely not anti-aliasing. Now it will check all the different pixels.

Reply:
We can do that, but I think the delta value has to be pretty high.
In the image below for example we would compare (222, 222, 222) and (11, 11, 11), which is antialiasing and should be checked.
![image](https://user-images.githubusercontent.com/7459510/115895633-ecb18180-a45a-11eb-9e33-50e0a738f317.png)

Do we have some kind of resource to know what the maximum delta difference in different antialiasing algorithms can be?